### PR TITLE
Improve Calendar. #1763 fixed

### PR DESCRIPTION
**Pull request by agolub-s**
### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#1763

### 💡 Background and solution

Improve user expirience with DatePicker.

1. Add commit changes after blur in input of DatePicker.
2. Fix click on suffix icon in DatePickerInput.

### 📝 Changelog

Add commit changes after blur in Calendar.
Fix click on suffix icon in DatePickerInput.

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Add commit changes after blur in Calendar. Fix click on suffix icon in DatePickerInput.     |
| 🇨🇳 Chinese |       在日历中模糊后添加提交更改。修复单击DatePicker输入中的后缀图标。   |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/date-picker/internal/DatePickerBase.cs [hunk 1]
```diff
@@ -186,7 +193,9 @@ public override CultureInfo CultureInfo
         public string DropdownClassName { get; set; }
 
         [Parameter]
-        public string Format { get; set; }
+        public OneOf<string, string[]> Format { get; set; }
```
Comment (ElderJames) [inline components/date-picker/internal/DatePickerBase.cs:196]: Only `RangePicker` can apply two `Format`, so please separate them, and do this by another PR.

Comment (agolub-s) [reply components/date-picker/internal/DatePickerBase.cs:196]: No, multiple formats can apply all pick date components, which inherits DatePickerBase. 
For your approve, have I to separate commits or...? I dont think so, because, I separated commits in one MR

P.S. I added additional examples for format after this message. You can check, DatePicker and RangeDatePicker

Comment (ElderJames) [reply components/date-picker/internal/DatePickerBase.cs:196]: The `Format` parameter is just a string for display, notinput.

And this PR would be squashed after merged.

Comment (agolub-s) [reply components/date-picker/internal/DatePickerBase.cs:196]: Okay, I'll separate this PR, but can't undestand why do you have documentation for "format" as "to set the date format, refer to [moment.js](http://momentjs.com/). When an array is provided, all values are used for parsing and first value is used for formatting.".
Do you whant to make two parameters: "DisplayFormat " and "InputFormats"?

Comment (agolub-s) [reply components/date-picker/internal/DatePickerBase.cs:196]: I think it is should be so, because, if we will have two parameters, we need to set this like
`DisplayFormat : "dd/MM/yyyy"`,
`InputFormat : "dd/MM/yyyy", "ddMMyyyy"`
I think, users want to change one or few digitals in a date, therefore it will be always duplicated for input string as `DisplayFormat`

Comment (agolub-s) [reply components/date-picker/internal/DatePickerBase.cs]: https://github.com/ant-design-blazor/ant-design-blazor/pull/3120

**Comment by agolub-s** [comment]
Hello, is anyone here?

**Comment by ElderJames** [comment]
Hello @agolub-s , please sign the CLA.

**Comment by agolub-s** [comment]
> @agolub-s please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @dotnet-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @dotnet-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @dotnet-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@dotnet-policy-service agree company="Quirco"

**Comment by agolub-s** [comment]
> Hello @agolub-s , please sign the CLA.

Hello, James.

I signed CLA. Thanks for help with my merge requests.
Should I to do something with it?

**Comment by agolub-s** [comment]
@ElderJames Hi, did you check it?

**Comment by ElderJames** [comment]
Yes,I will. The tests was fail and need to be fixed.

**Comment by agolub-s** [comment]
@ElderJames Hi, sorry for my late, I fixed tests. And can you check this request, please, https://github.com/ant-design-blazor/ant-design-blazor/pull/3120

**Comment by SumitAngra** [comment]
@agolub-s Is there anything pending on this PR to complete ?

**Comment by agolub-s** [comment]
> @agolub-s Is there anything pending on this PR to complete ?

I don't know(

File: components/date-picker/DatePicker.razor [hunk 2]
```diff
@@ -36,7 +36,7 @@
                                  AutoFocus="@AutoFocus"
                                  OnClick="async e => { await OnInputClick(); }"
                                  OnSuffixIconClick="OnSuffixIconClick"
-                                 OnKeyDown="@OnKeyDown"
+                                 OnKeyDown="@(e => OnKeyDown(e))"
```
Comment (ElderJames) [inline components/date-picker/DatePicker.razor:39]: Here is no need to change

File: components/date-picker/RangePicker.razor.cs [hunk 3]
```diff
@@ -453,7 +474,7 @@ public override void ChangeValue(DateTime value, int index = 0, bool closeDropdo
                 return;
             }
 
-            bool isValueInstantiated = Value == null;
+            var isValueInstantiated = Value == null;
```
Comment (ElderJames) [inline components/date-picker/RangePicker.razor.cs:477]: Here is no need to change

File: components/date-picker/RangePicker.razor.cs [hunk 4]
```diff
@@ -509,7 +530,7 @@ public override void ClearValue(int index = -1, bool closeDropdown = true)
             _isSetPicker = false;
 
             var array = CurrentValue as Array;
-            int[] indexToClear = index == -1 ? new[] { 0, 1 } : new[] { index };
+            var indexToClear = index == -1 ? new[] { 0, 1 } : new[] { index };
```
Comment (ElderJames) [inline components/date-picker/RangePicker.razor.cs:533]: Here is no need to change

**Comment by SumitAngra** [comment]
@agolub-s Thanks for your response.
I can see a few review comments on the PR added by james yesterday :)
Hoping to get this PR merged soon!

**Comment by agolub-s** [comment]
> @agolub-s Is there anything pending on this PR to complete ?

Thank you so much, I waiting this merge request too much, I hope it will be soon)

**Comment by ElderJames** [comment]
@agolub-s Thank you, this PR is ready to merge.

**Review by ElderJames — APPROVED** [review]
LGTM.
